### PR TITLE
feat(lib): add range selection behavior

### PR DESCRIPTION
**Pull request by d3lm**
This feature addresses #62 and adds range selection functionality to the library. This means that when an item is clicked, this item is marked as the start item for a range selection. The next item that is clicked by holding down `shift` will select all items in between. This is now the default for `shift`.

### CODE REVIEW

**Comment by user23022** [comment]
Shall I wait with checking this until your build is successful?

**Comment by d3lm** [comment]
@user23022 No, you can go ahead and check it out. I am trying to find out why the build fails and in other PRs it works fine.

**Comment by user23022** [comment]
Thanks, will check it out when I have some time

**Comment by d3lm** [comment]
Yay, the build finally succeeds. Travis is hard 😂

File: projects/ngx-drag-to-select/src/lib/select-container.component.ts [hunk 1]
```diff
@@ -410,21 +421,61 @@ export class SelectContainerComponent implements AfterViewInit, OnDestroy {
...
+    }
+
+    if (!this.shortcuts.extendedSelectionShortcut(event) && currentIndex === -1) {
+      if (this._lastStartIndex >= 0) {
+        const lastStart = this._selectableItems[this._lastStartIndex];
+        lastStart.toggleRangeStart();
+      }
+
+      this._lastStartIndex = -1;
+    }
```
Comment (timdeschryver) [inline projects/ngx-drag-to-select/src/lib/select-container.component.ts]:
```ts
if (!this.shortcuts.extendedSelectionShortcut(event)) {
     const lastRangeStart = this._selectableItems[this._lastStartIndex];
      if (lastRangeStart) {
        lastRangeStart.toggleRangeStart();
      }

    this._lastStartIndex = currentIndex;
    if(this._lastStartIndex  !== 1) { 
      clickedItem.toggleRangeStart();
   }
} 
```

OR via a setter?
The below is pseudo-code
```ts
private _lastStartIndex set(index: number | undefined) {
   if(previousLastStartIndex) {
     items[previousLastStartIndex].toggle()
   }


  items[index].toggle();
  previousLastStartIndex = index;
}

```

Comment (d3lm) [reply projects/ngx-drag-to-select/src/lib/select-container.component.ts]: Yep, thanks! I have just pushed my changes. It refactors this particular piece as well. Nice spotting 👍

**Comment by user23022** [comment]
I notice that the behavior is different than on my Ubuntu folder.
In your app you can Ctrl + click some items to make a custom selection, then Shift + click will add the range on top of the selection. Where as in Ubuntu, if you Shift + click, your previous selection is always deselected when you perform a Shift + click.

I get that it might be useful to select a range after selecting items one by one.
The only weird thing is that if you perform a range selection with Click, Shift + Click > Then Ctrl + click to deselect one in the range > Then shift + click some more (further than the last item you clicked before) the item you deselected with Ctrl + click will be selected again. The range added to your selection will start from that point.

This might be the reason for Ubuntu to just keep shift + click functionality simple. Weird edge cases (from a UX perspective) are more common if you make it more complex.

**Comment by d3lm** [comment]
@user23022 Thanks for your awesome feedback and testing out this feature!

I intentionally implemented it this way to make it more convenient to have multiple range selections. You can even `Shift + R` to move the range start around without selecting an item. I think this can be quite handy and I wanted to make it fit more complex scenarios as well, cause I figured that range selection is a really essential feature. But I do get it that Ubuntu has a rather simple range selection mechanism. This makes the implementation more lightweight for sure.

So if you wanna simply deselect your previous range selection you can just select one item without pressing `Ctrl`. This will deselect all previously selected items and start a completely new selection, whereas if you hold down `Ctrl` I assume that you want to add on top of your previous selection. That's already how this library worked before.

Would it work for you if I merged it the way it is now? I am quite happy with it because it's more advanced and more flexible IMO.

In that case, I ll go ahead and merge it and cut a release 🙌

**Comment by d3lm** [comment]
Alright, just pushed my final changes. As part of this PR I have also improved the shortcuts so that they can also contain any alphabetic character ranging from `a - z`, and not only "meta" keys 🙌This is a small but cool addition I think.

**Comment by d3lm** [comment]
It's now also possible to disable the range selection feature.

**Pull request merged**